test/models/patient_duplicate_detection_test.rb

```diff
@@ -0,0 +1,187 @@
+                                            patient_dup[:user_defined_id_statelocal])
+
+    assert duplicate_data[:is_duplicate]
+    assert_equal(duplicate_data[:duplicate_field_data], [
+                   {
+                     count: 14,
+                     fields: ['First Name', 'Last Name']
+                   }
+                 ])
+  end
+
+  test 'duplicate_data finds duplicate with input of FN ID' do
+    patient_dup = Patient.new(first_name: 'Deckard', last_name: nil, date_of_birth: nil, sex: nil, user_defined_id_statelocal: 'asdf123')
+
+    duplicate_data = Patient.duplicate_data(patient_dup[:first_name],
+                                            patient_dup[:last_name],
+                                            patient_dup[:sex],
+                                            patient_dup[:date_of_birth]&.strftime('%F'),
+                                            patient_dup[:user_defined_id_statelocal])
+
+    assert duplicate_data[:is_duplicate]
+    assert_equal(duplicate_data[:duplicate_field_data], [
+                   {
+                     count: 1,
+                     fields: ['State/Local ID']
+                   }
+                 ])
+  end
+
+  test 'duplicate_data finds duplicate with input of LN ID' do
+    patient_dup = Patient.new(first_name: nil, last_name: 'Cain', date_of_birth: nil, sex: nil, user_defined_id_statelocal: 'asdf123')
+
+    duplicate_data = Patient.duplicate_data(patient_dup[:first_name],
+                                            patient_dup[:last_name],
+                                            patient_dup[:sex],
+                                            patient_dup[:date_of_birth]&.strftime('%F'),
+                                            patient_dup[:user_defined_id_statelocal])
+
+    assert duplicate_data[:is_duplicate]
+    assert_equal(duplicate_data[:duplicate_field_data], [
+                   {
+                     count: 1,
+                     fields: ['State/Local ID']
+                   }
+                 ])
+  end
+
+  test 'duplicate_data finds duplicate with input of ID' do
+    patient_dup = Patient.new(first_name: nil, last_name: nil, date_of_birth: nil, sex: nil, user_defined_id_statelocal: 'asdf123')
+
+    duplicate_data = Patient.duplicate_data(patient_dup[:first_name],
+                                            patient_dup[:last_name],
+                                            patient_dup[:sex],
+                                            patient_dup[:date_of_birth]&.strftime('%F'),
+                                            patient_dup[:user_defined_id_statelocal])
+
+    assert duplicate_data[:is_duplicate]
+    assert_equal(duplicate_data[:duplicate_field_data], [
+                   {
+                     count: 1,
+                     fields: ['State/Local ID']
+                   }
+                 ])
+  end
+
+  test 'duplicate_data does NOT find duplicate with input of FN LN Sex DoB ID' do
```

Comment:
can we add more tests for cases where it's NOT a duplicate? if we add all these examples as test cases that Kate from the PO team outlined [here](https://teams.microsoft.com/l/file/6DF289BD-F078-4F36-8363-A789DC0459D7?tenantId=c620dc48-1d50-4952-8b39-df4d54d74d82&fileType=pptx&objectUrl=https%3A%2F%2Fmitre.sharepoint.com%2Fsites%2FCoVDT%2FShared%20Documents%2FGeneral%2FProduct%20Owner%20Materials%2FRequirements%2FSara%20Alert%20Duplicate%20Examples.pptx&baseUrl=https%3A%2F%2Fmitre.sharepoint.com%2Fsites%2FCoVDT&serviceName=teams&threadId=19:00d3d45c64ca457795d58c2764d517cd@thread.skype&groupId=7de0f318-3170-4f1c-82eb-95bfe132bfe9), we should be in good shape

Reply:
They should all be covered, I think. The return values are filtering out non matches from our "database" of 16 potential matches. In this file Line 19 changes sex, Line 20 changes DoB, Line 22 changes first name, Line 23 changes last name. We import a nil FN line 140 and LN  line 122. 

Did I miss a case?

Reply:
ah, you are right, thanks for clarifying!